src/NServiceBus.Core/Pipeline/Incoming/IHandlingMetrics.cs

```diff
@@ -0,0 +1,24 @@
+﻿namespace NServiceBus;
+
+using System;
+
+/// <summary>
+/// Registers the metrics related to a specific message handler execution.
+/// </summary>
+interface IHandlingMetrics
```

Comment:
To be more explicit about what metric we’re referring to here, I’d rename this type to `IMessageHandlingMetrics`